lib/graphql/schema/object.rb

```diff
@@ -69,64 +71,77 @@ def initialize(object, context)
       end
 
       class << self
-        def implements(*new_interfaces)
+        def implements(*new_interfaces, **options)
+          new_memberships = []
           new_interfaces.each do |int|
             if int.is_a?(Module)
               unless int.include?(GraphQL::Schema::Interface)
                 raise "#{int} cannot be implemented since it's not a GraphQL Interface. Use `include` for plain Ruby modules."
               end
 
+              new_memberships << int.type_membership_class.new(int, self, options)
+
               # Include the methods here,
               # `.fields` will use the inheritance chain
               # to find inherited fields
               include(int)
+            elsif int.is_a?(String) || int.is_a?(GraphQL::Schema::LateBoundType)
+              if options.any?
+                raise ArgumentError, "`implements(...)` doesn't support options with late-loaded types yet. Remove #{options} and open an issue to request this feature."
+              end
+              new_memberships << int
+            else
+              raise ArgumentError, "Unexpected interface definition (expected module): #{int} (#{int.class})"
```

Comment:
TODO This is a breaking change -- can I still accept legacy interfaces here? Otherwise I'll slate this for 1.11

Reply:
Fixed -- added a case for legacy interfaces.